astroid/tests/unittest_python3.py

```diff
@@ -258,7 +258,74 @@ def test_underscores_in_numeral_literal(self):
             self.assertIsInstance(inferred, nodes.Const)
             self.assertEqual(inferred.value, expected)
 
+    @require_version('3.6')
+    def test_async_comprehensions(self):
+        return_async_comp_funcs = [
+            extract_node("async def f(): return [i async for i in aiter() if i % 2]"),
+            extract_node("async def f(): return {i async for i in aiter() if i % 2}"),
+            extract_node("async def f(): return (i async for i in aiter() if i % 2)"),
+            extract_node("async def f(): return {i: i async for i in aiter() if i % 2}")
+        ]
+        return_non_async_comp_funcs = [
+            extract_node("async def f(): return {i: i for i in iter() if i % 2}")
+        ]
+
+        for func in return_async_comp_funcs:
```

Comment:
You can use ``@`` to extract a particular node, instead of doing it manually, such as in this example:

```
f = extract_node('''
async def f():
   return [i async for i in...] #@
''')
# now f is the Return
```
Or you can use ``__()`` for a fine grained approach, as in:

```
f = extract_node('''
async def f():
   return __([i async for i in...])
''')
# now f is the comprehension
```